strapi/strapi
# Fixes private field of target relation being reset - #9712

### What does it do?

This PR fixes issue #9712 

It preserves private field by setting `private: targetAttribute.private || undefined,` from `generateRelation` from content-type-builder.

### Why is it needed?

`private` of target relation content type is not preserved when editing content type.

### How to test it?

1. Create two content types -  Article, Comment
2. Set Relationship Article 1 : N Comment
3. Set Article.comments to private field
4. Set Comment.article to private field
5. Check `private` value of both content types are `true`


### Related issue(s)/PR(s)

#9712
